podcasts/Bookmarks/Episode/BookmarkEpisodeListController.swift

```diff
@@ -0,0 +1,52 @@
+import Combine
+import PocketCastsDataModel
+import SwiftUI
+
+class BookmarkEpisodeListController: ThemedHostingController<BookmarkEpisodeListView> {
+    private let playbackManager: PlaybackManager
+    private let bookmarkManager: BookmarkManager
+    private let viewModel: BookmarkEpisodeListViewModel
+
+    private var cancellables = Set<AnyCancellable>()
+
+    init(episode: BaseEpisode,
+         bookmarkManager: BookmarkManager = PlaybackManager.shared.bookmarkManager,
+         playbackManager: PlaybackManager = .shared) {
+
+        self.bookmarkManager = bookmarkManager
+        self.playbackManager = playbackManager
+
+        self.viewModel = BookmarkEpisodeListViewModel(episode: episode,
+                                                      bookmarkManager: bookmarkManager,
+                                                      sortOption: Constants.UserDefaults.bookmarks.podcastSort)
+
+        super.init(rootView: BookmarkEpisodeListView(viewModel: viewModel))
+
+        viewModel.router = self
+    }
+
+    @MainActor required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+// MARK: - BookmarkListRouter
+
+extension BookmarkEpisodeListController: BookmarkListRouter {
+    func bookmarkPlay(_ bookmark: Bookmark) {
+        playbackManager.playBookmark(bookmark)
```

Comment:
As the places of bookmarks grow I wonder if there might be a better solution for this?

We've been adding code to play/edit in all 3 places that can display Bookmarks, and I'm pretty sure this list will grow in the future. 😬

At this point, it seems that this is a logic that maybe the bookmark view model should handle, or maybe even a helper we can inject into the view (and in this dependency we can inject the `UIViewController`).

Thoughts?

Reply:
I agree, I can look into changing that in a future PR.